pytest-httpbin test failed due to override in octoprint

### Describe the bug

The recent nixos-unstable update (48d63e924a2666baf37f4f14a18f19347fbd54a2) breaks octoprint. Last known good commit is 98bb5b77c8c6666824a4c13d23befa1e07210ef1. 

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix-build -A octoprint  https://github.com/NixOS/nixpkgs/archive/48d63e924a2666baf37f4f14a18f19347fbd54a2.tar.gz`

result: 
```
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
tests/test_httpbin.py ...........127.0.0.1 - - [13/Feb/2022 16:36:33] "GET /get HTTP/1.1" 200 222
..                                      [ 65%]
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
pytestCheckPhase
Executing pytestCheckPhase
Installing collected packages: moto
============================= test session starts ==============================
platform linux -- Python 3.9.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /build/trio-0.19.0, configfile: pyproject.toml
Successfully installed moto-3.0.2
/build/moto-3.0.2
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/qzn94mcph0n4xg0k4w9b7jgqzigkcr6w-python3.9-moto-3.0.2
collected 610 items / 19 deselected / 591 selected

tests/test_server.py ......F                                             [100%]

=================================== FAILURES ===================================
______________ test_redirect_location_is_https_for_secure_server _______________

httpbin_secure = <pytest_httpbin.serve.SecureServer object at 0x7ffff570aa90>

    def test_redirect_location_is_https_for_secure_server(httpbin_secure):
        assert httpbin_secure.url.startswith('https://')
        response = requests.get(
            httpbin_secure + "/redirect-to?url=/html",
            allow_redirects=False
        )
        assert response.status_code == 302
        assert response.headers.get('Location')
>       assert response.headers['Location'].startswith('https://')
E       AssertionError: assert False
E        +  where False = <built-in method startswith of str object at 0x7ffff56ef270>('https://')
E        +    where <built-in method startswith of str object at 0x7ffff56ef270> = '/html'.startswith

tests/test_server.py:105: AssertionError
----------------------------- Captured stderr call -----------------------------
127.0.0.1 - - [13/Feb/2022 16:36:34] "GET /redirect-to?url=/html HTTP/1.1" 302 0
=========================== short test summary info ============================
FAILED tests/test_server.py::test_redirect_location_is_https_for_secure_server
========================= 1 failed, 19 passed in 2.26s =========================
error: builder for '/nix/store/43m4qchmrdf2i82vvsisj3nisp7khw0c-python3.9-pytest-httpbin-1.0.1.drv' failed with exit code 1;
       last 10 log lines:
       > E       AssertionError: assert False
       > E        +  where False = <built-in method startswith of str object at 0x7ffff56ef270>('https://')
       > E        +    where <built-in method startswith of str object at 0x7ffff56ef270> = '/html'.startswith
       >
       > tests/test_server.py:105: AssertionError
       > ----------------------------- Captured stderr call -----------------------------
       > 127.0.0.1 - - [13/Feb/2022 16:36:34] "GET /redirect-to?url=/html HTTP/1.1" 302 0
       > =========================== short test summary info ============================
       > FAILED tests/test_server.py::test_redirect_location_is_https_for_secure_server
       > ========================= 1 failed, 19 passed in 2.26s =========================
       For full logs, run 'nix log /nix/store/43m4qchmrdf2i82vvsisj3nisp7khw0c-python3.9-pytest-httpbin-1.0.1.drv'.
error: 1 dependencies of derivation '/nix/store/akwcam9k3ycvf5iws9mic238naz35k1r-python3.9-httpcore-0.14.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1p42r57cyldyx0fxqr118y4ch0pf30bb-python3.9-httpx-0.21.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6krp90zp3842ih9r6afmmif0rndggxwh-python3.9-sentry-sdk-1.5.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n1dcisc3fn9xpmgn6bpxmpslhvqbpgwg-python3.9-OctoPrint-1.7.3.drv' failed to build
```
This is probably because the derivation of octoprint has an override for `werkzeug`.  Werkzeug changed its API in version 2, which lead to a similar [issue in httpbin](https://github.com/postmanlabs/httpbin/issues/647). When `pytest-httpbin` v1.0.1 [introduced a new test](https://github.com/kevin1024/pytest-httpbin/commit/83f637dc932c3c2d67eae63f9c3818119abfc26f), it fails with the old Werkzeug version due to an incorrect redirect. This is also why `pytest-httpbin` builds without issues, but fails to build, when its needed for octoprint. 

I'm not a 100% sure its the werkzeug override. Octoprint has a lot of Python overrides..

### Expected behavior
build sucessfully.

### Notify maintainers
@abbradar @gebner @WhittlesJr 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 5.10.98, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/misc/octoprint/default.nix
+++ b/pkgs/applications/misc/octoprint/default.nix
@@ -6,24 +6,24 @@
 , substituteAll
 , nix-update-script
   # To include additional plugins, pass them here as an overlay.
-, packageOverrides ? self: super: {}
+, packageOverrides ? self: super: { }
 }:
 let
   mkOverride = attrname: version: sha256:
-  self: super: {
-    ${attrname} = super.${attrname}.overridePythonAttrs (
-      oldAttrs: {
-        inherit version;
-        src = oldAttrs.src.override {
-          inherit version sha256;
-        };
-      }
-    );
-  };
+    self: super: {
+      ${attrname} = super.${attrname}.overridePythonAttrs (
+        oldAttrs: {
+          inherit version;
+          src = oldAttrs.src.override {
+            inherit version sha256;
+          };
+        }
+      );
+    };
 
   py = python3.override {
     self = py;
-    packageOverrides = lib.foldr lib.composeExtensions (self: super: {}) (
+    packageOverrides = lib.foldr lib.composeExtensions (self: super: { }) (
       [
         # the following dependencies are non trivial to update since later versions introduce backwards incompatible
         # changes that might affect plugins, or due to other observed problems
@@ -57,7 +57,7 @@ let
                 inherit version;
                 sha256 = "6c80b1e5ad3665290ea39320b91e1be1e0d5f60652b964a3070216de83d2e47c";
               };
-              doCheck= false;
+              doCheck = false;
             });
           }
         )
@@ -92,7 +92,7 @@ let
                   pysocks
                 ];
                 disabledTests = [
-                  "testConnect"  # requires network access
+                  "testConnect" # requires network access
                 ];
               }
             );
@@ -154,10 +154,60 @@ let
               disabledTests = [
                 "test_apply_simulates_delivery_info"
                 "test_auto_enabling_integrations_catches_import_error"
+                "test_leaks"
               ];
               disabledTestPaths = [
                 # Don't test integrations
                 "tests/integrations"
+                # test crashes on aarch64
+                "tests/test_transport.py"
+              ];
+            });
+          }
+        )
+
+        # Octoprint fails due to a newly added test in pytest-httpbin
+        # see https://github.com/NixOS/nixpkgs/issues/159864
+        (
+          self: super: {
+            pytest-httpbin = super.pytest-httpbin.overridePythonAttrs (oldAttrs: rec {
+              disabledTests = [
+                "test_redirect_location_is_https_for_secure_server"
+              ];
+            });
+          }
+        )
+
+        # All test fail on aarch64
+        (
+          self: super: {
+            azure-core = super.azure-core.overridePythonAttrs (oldAttrs: rec {
+              doCheck = stdenv.buildPlatform == "x86_64-linux";
+            });
+          }
+        )
+
+        # needs network
+        (
+          self: super: {
+            falcon = super.falcon.overridePythonAttrs (oldAttrs: rec {
+              #pytestFlagsArray = [ "-W ignore::DeprecationWarning" ];
+              disabledTestPaths = oldAttrs.disabledTestPaths ++ [
+                "tests/asgi/test_asgi_servers.py"
+              ];
+            });
+          }
+        )
+
+        # update broke some tests
+        (
+          self: super: {
+            sanic = super.sanic.overridePythonAttrs (oldAttrs: rec {
+              disabledTestPaths = oldAttrs.disabledTestPaths ++ [
+                "test_cli.py"
+                "test_cookies.py"
+                # requires network
+                "test_worker.py"
               ];
             });
           }
@@ -302,29 +352,30 @@ let
                 })
               ];
 
-              postPatch = let
-                ignoreVersionConstraints = [
-                  "cachelib"
-                  "colorlog"
-                  "emoji"
-                  "immutabledict"
-                  "PyYAML"
-                  "sarge"
-                  "sentry-sdk"
-                  "watchdog"
-                  "wrapt"
-                  "zeroconf"
-                ];
-              in
+              postPatch =
+                let
+                  ignoreVersionConstraints = [
+                    "cachelib"
+                    "colorlog"
+                    "emoji"
+                    "immutabledict"
+                    "PyYAML"
+                    "sarge"
+                    "sentry-sdk"
+                    "watchdog"
+                    "wrapt"
+                    "zeroconf"
+                  ];
+                in
                 ''
-                  sed -r -i \
-                    ${lib.concatStringsSep "\n" (
-                  map (
-                    e:
-                      ''-e 's@${e}[<>=]+.*@${e}",@g' \''
-                  ) ignoreVersionConstraints
-                )}
-                    setup.py
+                    sed -r -i \
+                      ${lib.concatStringsSep "\n" (
+                    map (
+                      e:
+                        ''-e 's@${e}[<>=]+.*@${e}",@g' \''
+                    ) ignoreVersionConstraints
+                  )}
+                      setup.py
                 '';
 
               dontUseSetuptoolsCheck = true;
@@ -360,4 +411,4 @@ let
     );
   };
 in
-  with py.pkgs; toPythonApplication octoprint
+with py.pkgs; toPythonApplication octoprint

